addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -336,17 +336,13 @@ export class ProductScreen extends Component {
     }
 
     getProductsBySearchWord(searchWord) {
-        const products = this.pos.selectedCategory?.id
-            ? this.getProductsByCategory(this.pos.selectedCategory)
-            : this.products;
-
-        const exactMatches = products.filter((product) => product.exactMatch(searchWord));
+        const exactMatches = this.products.filter((product) => product.exactMatch(searchWord));
 
         if (exactMatches.length > 0 && searchWord.length > 2) {
             return exactMatches;
         }
 
-        const fuzzyMatches = fuzzyLookup(unaccent(searchWord, false), products, (product) =>
+        const fuzzyMatches = fuzzyLookup(unaccent(searchWord, false), this.products, (product) =>
```

Comment:
could you check what the results of the search look like if you remove the `exactMatches` logic and do the fuzzy search directly ? The fuzzy search algorithm should give more weight to exact matches so maybe the result would be similar.  That would help in reducing some code.